internal/provider/utils.go

```diff
@@ -42,9 +42,16 @@ func isInStr(l []string, i string) bool {
 }
 
 func IsValidContract(sub *substrate.Substrate, contractID uint64) (bool, error) {
```

Comment:
please move to contract_management.go file and keep it in the same package no problem, because utils, helpers .. etc don't really add any meaning to the file and indicate nothing about the functionality